docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "How to convert an existing Gatsby blog to use MDX"
+date: 2019-11-10
+author: Thomas Wang
+excerpt: "It can be a big tricky to add MDX to an existing blog. This blog post covers how to install and configure MDX to Gatsby's blog starter"
+tags:
+  - blogs
+  - mdx
+  - markdown
+---
+
+## What is MDX?
+
+MDX is a file format that lets you add JSX to Markdown files via a `.mdx` file! It was started by and is maintained by Gatsby's [John Otander](https://github.com/johno) and the [open source community on Github](https://github.com/mdx-js/mdx).
+
+With MDX, you can import React components and declare them alongside regular markdown syntax. This is useful to display specific component examples alongside the actual code, complex interactive components like charts, and reusable components across multiple pages like sign-up forms.
+
+## Adding MDX to Gatsby Starter Blog
+
+It can be a big tricky to add MDX to an existing blog. The following 5 steps will cover how to install and configure MDX to [Gatsby's blog starter](https://github.com/gatsbyjs/gatsby-starter-blog/tree/master), which as of [today's version](https://github.com/gatsbyjs/gatsby-starter-blog/tree/8852c1e51651b902f19706ff2ca9f60dabc25709), does not have MDX pre-installed.
+
+You can also see the [full changes in PR #18929](https://github.com/gatsbyjs/gatsby/pull/18928/files) for an overview of the changes you have to make to get MDX working.
+
+### Step 1
+
+Install [gatsby-plugin-mdx](/packages/gatsby-plugin-mdx/), the official plugin for using MDX with Gatsby. Also, install `@mdx-js/mdx` and `@mdx-js/react`.
+
+```
+npm install --save gatsby-plugin-mdx @mdx-js/mdx @mdx-js/react
+```
+
+### Step 2
+
+In `gatsby-config.js`, replace `gatsby-transformer-remark` with `gatsby-plugin-mdx`, along with the following changes:
+
+```diff:title=gatsby-config.js
```

Comment:
I'm not sure a diff is the way to go here. I worry that having it be multiple lines will confuse the reader. What do we think about just showing the full `gatsby-plugin-mdx` configuration and ignoring what does and doesn't overlap with `gatsby-transformer-remark`?

Reply:
if we post the full config then the reader has to look over the code to find out what are the differences :( (see [first version of this file](https://github.com/gatsbyjs/gatsby/blob/147454d1a32a982f2591e836cfb74b28d64cbf0f/docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md) with full block diff - but then it's hard to find the important parts)

Reply:
I've seen diffs like this in the docs before, but yeah that's what I had originally. I thought the change made sense :)

Reply:
As an experienced developer I really like the diff (it's a power tool for me that allows me to move faster), but I can also understand that less experienced developers may not understand this right away (or at all).

Maybe we can provide tools for both levels of understanding? Diff for the power users, and then a written example with code snippets afterwards, showing and explaining the changes that are highlighted in the diff?

I see this kind of like "progressive enhancement" for developer documentation :)

Reply:
I'm a big fan of diffs as well, but there are a couple things about this one that strike me as more complex to follow. In particular, it's multiple lines that aren't consecutive, it's altering information and adding it at the same time, but the + symbol will always be the same.
I worry that showing both won't really help the newer users.
Do we actually think it harms power users to have the non-diff version as the default? I'd rather opt for the version that applies to everyone if we can.

Reply:
Because the topic is "changing to" here would better a diff. Because every has to apply the changes individual by his setup which would be very different from the starter blog. 

I agree in other guides a copy paste is better. 

But as A solution to fit both way we can add the 1-6lines on top with the changes as diff and an full codesnippet as complete for copy and paste

Reply:
That's fair and something I hadn't considered. The idea that people will have projects with different configs over time. Seems like a reasonable compromise to handle both situations.

Reply:
done